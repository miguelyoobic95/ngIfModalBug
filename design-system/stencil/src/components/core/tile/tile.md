---
name: Tile
category: Core Components
---

```tile.html
    <yoo-tile id="tile1" text="ONGOING MISSIONS" value="5"></yoo-tile>
    <br/><br/>
    <yoo-tile id="tile2" text="PENDING MISSIONS" value="7"></yoo-tile>
```

```tile.js hidden
  var comp = document.querySelector('#tile1');
  comp.textClass = "info";

  var comp2 = document.querySelector('#tile2');
  comp2.textClass = "warning";
```

```tile-example.html
    <yoo-tile icon="yo-activity" text="ONGOING MISSIONS" textClass="info" value="5"></yoo-tile>
```

## Attributes

|Name|Type|Default|Description|
|---|---|---|---|
|`icon`|string|   |the icon displayed in the tile|
|`value`|string|   |the value displayed in the tile|
|`text`|string|   |the text displayed in the tile|
|`textClass`|string|   |the css of the text|

## CSS

|Type|Name|Description|
|---|---|---|
|Accent|`accent`|tile accent|
|Dark|`dark`|tile dark|
|Success|`success`|tile success|
|Danger|`danger`|tile danger|
|Info|`info`|tile info|
|Warning|`warning`|tile warning|
|Gradient Accent|`gradient-accent`|tile gradient-accent|
|Gradient Dark|`gradient-dark`|tile gradient-grey|
|Gradient Success|`gradient-success`|tile gradient-success|
|Gradient Danger|`gradient-danger`|tile gradient-danger|
|Gradient Info|`gradient-info`|tile gradient-info|
|Gradient Warning|`gradient-warning`|tile gradient-warning|