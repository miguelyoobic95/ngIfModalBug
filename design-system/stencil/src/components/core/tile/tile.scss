:host {
  .outer-container {
    display: flex;
    flex-direction: column;
    justify-content: flex-end;

    width: 10.3125rem;
    height: 8.125rem;
    background-color: cssvar(light);
    box-shadow: $tile-shadow;
    border-radius: 0.5rem;

    padding: 0.96875rem 1.25rem;
 
    .tile-icon {
      // margin-bottom: 0.9375rem;
    }
    .tile-text {
      font-size: 0.625rem;
      margin-top: 0.34375rem;
      line-height: 0.75rem;
    }
    .tile-value {
      font-size: 2.25rem;
      font-weight: 300;
      line-height: 2.75rem;
    }
  }
}
