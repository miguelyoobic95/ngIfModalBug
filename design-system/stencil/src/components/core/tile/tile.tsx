import { Component, Element, Prop } from '@stencil/core';

@Component({
    tag: 'yoo-tile',
    styleUrl: 'tile.scss',
    scoped: true
})
export class YooTileComponent {

    @Prop() icon: string;
    @Prop() value: string;
    @Prop() text: string;
    @Prop() textClass: string;

     @Element() host: HTMLStencilElement;

    render(): JSX.Element {
        return (
            <div class="outer-container">
                {(this.icon ? <div class="tile-icon"><i class={this.icon}></i></div> : null )}
                <div class="tile-value">{this.value}</div>
                <div class={'tile-text ' + (this.textClass)}>{this.text}</div>
            </div>
        );
    }
}
