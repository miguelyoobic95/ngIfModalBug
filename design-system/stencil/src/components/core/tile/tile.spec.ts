import { TestWindow } from '@stencil/core/testing';
import { YooTileComponent } from './tile';

describe('TileComponent', () => {
    it('should build', () => {
        expect(new YooTileComponent()).toBeTruthy();
    });

    describe('Rendering', () => {
        let element, window;
        beforeEach( async () => {
            window = new TestWindow();
            element = await window.load({
                components: [YooTileComponent],
                html: '<yoo-tile></yoo-tile>'
            });
        });

        it ('Should render', async () => {
            expect(element).toMatchSnapshot();
        });
        it('Should render with text', async () => {
            element.text = 'ONGOING MISSIONS';
            await window.flush();
            expect(element).toMatchSnapshot();
        });
        it('Should render with textClass', async () => {
            element.textClass = 'gradient-success';
            await window.flush();
            expect(element).toMatchSnapshot();
        });
        it('Should render with value', async () => {
            element.value = '5';
            await window.flush();
            expect(element).toMatchSnapshot();
        });
        it('Should render with icon', async () => {
            element.icon = 'yo-activity';
            await window.flush();
            expect(element).toMatchSnapshot();
        });
    });
    // describe('Events', () => {

    //     let emitEvent, window;
    //     beforeEach( async () => {
    //         window = new TestWindow();
    //         emitEvent = jest.fn();
    //     });
    // });
});