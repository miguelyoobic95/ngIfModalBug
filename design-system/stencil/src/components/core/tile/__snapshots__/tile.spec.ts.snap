// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TileComponent Rendering Should render 1`] = `
<yoo-tile
  class="hydrated"
  data-yoo-tile-host=""
>
  
  <div
    class="outer-container"
    data-yoo-tile=""
  >
    
    <div
      class="tile-value"
      data-yoo-tile=""
    >
      
    </div>
    <div
      class="tile-text undefined"
      data-yoo-tile=""
    >
      
    </div>
  </div>
</yoo-tile>
`;

exports[`TileComponent Rendering Should render with icon 1`] = `
<yoo-tile
  class="hydrated"
  data-yoo-tile-host=""
>
  
  <div
    class="outer-container"
    data-yoo-tile=""
  >
    <div
      class="tile-icon"
      data-yoo-tile=""
    >
      <i
        class="yo-activity"
        data-yoo-tile=""
      />
    </div>
    <div
      class="tile-value"
      data-yoo-tile=""
    >
      
    </div>
    <div
      class="tile-text undefined"
      data-yoo-tile=""
    >
      
    </div>
  </div>
</yoo-tile>
`;

exports[`TileComponent Rendering Should render with text 1`] = `
<yoo-tile
  class="hydrated"
  data-yoo-tile-host=""
>
  
  <div
    class="outer-container"
    data-yoo-tile=""
  >
    
    <div
      class="tile-value"
      data-yoo-tile=""
    >
      
    </div>
    <div
      class="tile-text undefined"
      data-yoo-tile=""
    >
      ONGOING MISSIONS
    </div>
  </div>
</yoo-tile>
`;

exports[`TileComponent Rendering Should render with textClass 1`] = `
<yoo-tile
  class="hydrated"
  data-yoo-tile-host=""
>
  
  <div
    class="outer-container"
    data-yoo-tile=""
  >
    
    <div
      class="tile-value"
      data-yoo-tile=""
    >
      
    </div>
    <div
      class="tile-text gradient-success"
      data-yoo-tile=""
    >
      
    </div>
  </div>
</yoo-tile>
`;

exports[`TileComponent Rendering Should render with value 1`] = `
<yoo-tile
  class="hydrated"
  data-yoo-tile-host=""
>
  
  <div
    class="outer-container"
    data-yoo-tile=""
  >
    
    <div
      class="tile-value"
      data-yoo-tile=""
    >
      5
    </div>
    <div
      class="tile-text undefined"
      data-yoo-tile=""
    >
      
    </div>
  </div>
</yoo-tile>
`;
